USAGE.md

```diff
@@ -1,30 +1,219 @@
 # Using the RAPIDS devcontainers
 
-### See the list of `rapidsai/devcontainers` tags [on DockerHub](https://hub.docker.com/r/rapidsai/devcontainers/tags).
+- [Using the RAPIDS devcontainers](#using-the-rapids-devcontainers)
+  - [System requirements](#system-requirements)
+  - [Quick start](#quick-start)
+  - [Detailed start](#detailed-start)
+    - [Using devcontainers in VS Code](#using-devcontainers-in-vs-code)
+    - [Using devcontainers from the terminal](#using-devcontainers-from-the-terminal)
+    - [Generated build scripts](#generated-build-scripts)
+    - [rapids-build-utils](#rapids-build-utils)
+    - [Native build tools - CMake, python builds](#native-build-tools---cmake-python-builds)
+    - [Exiting the devcontainer](#exiting-the-devcontainer)
+  - [Adding projects: `manifest.yaml` file](#adding-projects-manifestyaml-file)
+  - [Using the pre-built images](#using-the-pre-built-images)
+    - [Using in `devcontainer.json`](#using-in-devcontainerjson)
+    - [Custom devcontainers](#custom-devcontainers)
+  - [Build caching with `sccache`](#build-caching-with-sccache)
+    - [Build caching with private S3 buckets](#build-caching-with-private-s3-buckets)
+    - [Using GitHub OAuth to issue S3 credentials via Hashicorp Vault](#using-github-oauth-to-issue-s3-credentials-via-hashicorp-vault)
 
-* [Using the pre-built images](#using-the-pre-built-images)
-  * [Using in `devcontainer.json`](#using-in-devcontainerjson)
-* [Build caching with sccache](#build-caching-with-sccache)
+## System requirements
 
+Devcontainers can be used on Linux, Mac and Windows. They use Docker, so the
+system requirements and limitations associated with Docker apply here also. On
+Mac and Windows, where a Linux VM must run to support Docker, you may need to be
+aware of memory limitations of that Linux VM. Otherwise, devcontainers do not add
+system requirements beyond the needs of the individual projects we're building.
+
+## Quick start
```

Comment:
comment: I'm biased, but I think the "Quick Start" section should assume someone is using VSCode. The whole point of devcontainers is making it stupid simple easy to get up and running. Having to `curl` and run some random script undermines that goal imo. 

Using the devcontainer from the CLI should be considered an advanced usage mode.

For reference, here is the equivalent documentation for CCCL's devcontainers: https://github.com/NVIDIA/cccl/blob/main/.devcontainer/README.md 

Anything that requires devcontainer-cli or

Reply:
I definitely understand that position. I share it to some extent. The difference I see is that VS Code requires talking through a GUI - telling people what to click on and such - which you have in the screenshots in CCCL's docs. In that way, it will always be harder than a copy-paste command to drop someone into an "activated environment." By "harder" I mean more steps to follow, which always has a cognitive load cost.

Switching between environments is also more intuitive on the CLI (exit container, open new one) than the whole "reopen in ..." dance, but I don't think that's terribly important.

Thanks for the pointer to your docs, though - there's a lot of good ideas that I'll add in here.

Reply:
> The difference I see is that VS Code requires talking through a GUI

We _could_ provide a button that gives you a [single click experience of launching VSCode with a particular devcontainer](https://code.visualstudio.com/docs/devcontainers/create-dev-container#_add-an-open-in-dev-container-badge), like this:
[![Open in Dev Containers](https://img.shields.io/static/v1?label=Dev%20Containers&message=Open&color=blue&logo=visualstudiocode)](https://vscode.dev/redirect?url=vscode://ms-vscode-remote.remote-containers/cloneInVolume?url=https://github.com/microsoft/vscode-remote-try-java)

Sadly, this doesn't work with our current devcontainers because we rely on the `workspaceMount` variable to set the working directory to `/home/coder` inside the container. The `cloneInVolume` URI functionality that makes that button work doesn't currently support `workspaceMount` [as discussed in this issue](https://github.com/microsoft/vscode-remote-release/issues/3034). 

[Codespaces](https://github.com/NVIDIA/cccl/blob/main/.devcontainer/README.md#github-codespaces) are about as close as we can get right now to providing that single-click experience. 

> Switching between environments is also more intuitive on the CLI 

I will definitely agree on this point. I've complained to the devcontainer devs that switching between containers in VSCode is very annoying.

Reply:
I agree with Jake. Launching devcontainers directly from the CLI is a more advanced workflow that involves more steps than clicking a button in VSCode.

I do think it's worthwhile to support that mode, but it should have feature-parity with the VSCode Dev Containers extension (and ideally use a [standalone @devcontainers/cli binary](https://github.com/devcontainers/cli/pull/343), not require someone to install node), and we should include the wrapper in this repo, not a random gist.

Reply:
This has been rearranged such that the VS Code approach comes first and is noted as recommended. It's in the USAGE_IN_PROJECTS.md file now.